forte/processors/data_augment/algorithms/word_splitting_processor.py

```diff
@@ -0,0 +1,122 @@
+            indexes: List[int] = []
+            endings = []
+            annos: Iterable[Annotation] = data_pack.get(augment_entry)
+            for idx, anno in enumerate(annos):
+                annotations.append(anno)
+                indexes.append(idx)
+                endings.append(anno.end)
+            if len(annotations) > 0:
+                annotation_to_split = random.sample(
+                    [
+                        (anno, idx)
+                        for (anno, idx) in zip(annotations, indexes)
+                        if (anno.end - anno.begin) > 1
+                    ],
+                    ceil(self.configs["alpha"] * len(annotations)),
+                )
+                annotation_to_split = sorted(
+                    annotation_to_split, key=lambda x: x[1], reverse=True
+                )
+                for curr_anno in annotation_to_split:
+                    src_anno, src_idx = curr_anno
+                    splitting_position = random.randrange(
+                        1, (src_anno.end - src_anno.begin)
+                    )
+                    word_split = [
+                        src_anno.text[:splitting_position],
+                        src_anno.text[splitting_position:],
+                    ]
+                    if src_idx != 0:
+                        first_position = endings[src_idx - 1] + 1
+                        second_position = endings[src_idx]
+                        word_split[1] = " " + word_split[1]
+                    else:
+                        first_position = 0
+                        second_position = endings[0]
+                        word_split[1] = " " + word_split[1]
+
+                    self._insert(word_split[1], data_pack, second_position)
+                    self._delete(src_anno)
+                    self._insert(word_split[0], data_pack, first_position)
+
+    @classmethod
+    def default_configs(cls):
+        """
+        Returns:
+        A dictionary with the default config for this processor.
+        Additional keys for determining how many words will be split:
+        - alpha: 0 <= alpha <= 1. indicates the percent of the words
+        in a sentence that are changed. The processor will perform
+        the Word Splitting operation 2 * (input length * alpha) times
+        after deleting the original annotation.
+        """
+        config = super().default_configs()
+        config.update(
+            {
+                "augment_entry": "ft.onto.base_ontology.Token",
+                "other_entry_policy": {
+                    "type": "",
+                    "kwargs": {
+                        "ft.onto.base_ontology.EntityMention": "auto_align",
+                        "ft.onto.base_ontology.Document": "auto_align",
+                        "ft.onto.base_ontology.Sentence": "auto_align",
+                    },
+                },
+                "data_aug_op": "forte.processors.data_augment.algorithms.",
```

Comment:
This `data_aug_op` does not point to a correct class name.

Reply:
Some parameters, like this `data_aug_op` is probably not even used in your class. Consider remove it?